src/tty/unix.rs

```diff
@@ -376,6 +428,33 @@ impl EventedReadWrite for Pty {
     fn write_token(&self) -> mio::Token {
         self.token
     }
+
+    #[cfg(unix)]
+    #[inline]
+    fn process_signals(&mut self) -> bool {
+        let sig = {
+            let mut sig = [0u8];
+            self.sig_pipe_read.read_exact(&mut sig).unwrap();
+            sig[0]
+        };
+
+        match sig as i32 {
+            libc::SIGCHLD => {
+                if let Some(_) = self.child.try_wait().expect("error collecting pty child") {
+                    return true;
+                }
```

Comment:
We have to be careful here to flat-out _ignore_ `SIGCHLD` from other sources, or we risk breaking anything that uses `spawn_daemon` or similar.